packages/cli/cloud/src/create-project/utils/project-questions.ts

```diff
@@ -0,0 +1,73 @@
+import { DistinctQuestion } from 'inquirer';
+import type { ProjectAnswers } from '../../types';
+
+/**
+ * Apply default values to questions based on the provided mapper
+ * @param QuestionsMapper - A partial object with keys matching the ProjectAnswers keys and values being the default value or a function to get the default value
+ */
+export function questionDefaultValuesMapper(
+  QuestionsMapper: Partial<{
+    [K in keyof ProjectAnswers]:
+      | ((question: DistinctQuestion<ProjectAnswers>) => ProjectAnswers[K])
+      | ProjectAnswers[K];
+  }>
+) {
+  return (
+    questions: ReadonlyArray<DistinctQuestion<ProjectAnswers>>
+  ): ReadonlyArray<DistinctQuestion<ProjectAnswers>> => {
+    return questions.map((question) => {
+      const questionName = question.name as keyof ProjectAnswers;
+
+      // If the question is part of the mapper, apply the default value
+      if (question.name && QuestionsMapper[questionName]) {
+        const questionDefault = QuestionsMapper[questionName];
+
+        // If the default value is a function, call it with the question and get the default value
+        if (typeof questionDefault === 'function') {
+          return {
+            ...question,
+            default: questionDefault(question),
+          };
+        }
+        // else we consider it as a static value
+        return {
+          ...question,
+          default: questionDefault,
+        };
+      }
+      return question;
+    });
+  };
+}
+
+/**
+ * Get default values from questions
+ * @param questions - An array of questions for project creation
+ */
+export function getDefaultsFromQuestions(
+  questions: ReadonlyArray<DistinctQuestion<ProjectAnswers>>
+) {
+  return questions.reduce((acc, question) => {
+    if (question.default && question.name) {
+      return { ...acc, [question.name]: question.default };
+    }
+    return acc;
+  }, {});
+}
+
+/**
+ * Get the default node version based on the current node version if it is in the list of choices
+ * @param question - The question for the node version in project creation
+ */
+export function getProjectNodeVersionDefault(question: DistinctQuestion<ProjectAnswers>) {
```

Comment:
(suggestion) Returned type would help here (I think it's simply a string?)

Reply:
Should be good now 👍